components/eam/src/physics/rrtmgp/radiation.F90

```diff
@@ -413,14 +441,71 @@ end function radiation_nextsw_cday
 
    !================================================================================================
 
-   subroutine radiation_init(state)
+   subroutine get_aerosol_optical_property_from_file(month_int,aerosol_optical_property_str,band_identifier,nbands,aerosol_optical_property)
+      use pio,            only: file_desc_t, pio_nowrite
+      use cam_pio_utils,    only: cam_pio_openfile,cam_pio_closefile
+      use cam_grid_support, only: cam_grid_check,cam_grid_id,cam_grid_get_dim_names
+      use ncdio_atm,       only: infld
+      use time_manager,   only: get_curr_date
+      use cam_logfile,     only: iulog
+
+      integer, intent(in) :: month_int, nbands !nbands is either nswbands or nlwbands
+      character*(*), intent(in) :: aerosol_optical_property_str, band_identifier
+      real(r8), intent(inout) :: aerosol_optical_property(pcols,pver,nbands,begchunk:endchunk)
+      !real(r8), intent(inout) :: aerosol_optical_property(pcol,pver,nbands)
+
+     !internal variables
+      character(len=100) :: base_file_name, optics_lookup_dir
+      character(len=500) :: filename
+      character(len=20) :: dim1name, dim2name
+      character(len=20) :: mon_str
+      type(file_desc_t) :: nccn_ncid
+      integer :: year, month, day, tod, next_month, grid_id
+      logical :: found = .false.
+
+      write(mon_str,*) month_int
+
+      !assign base_file_name the name of the CCN file being used 
+      base_file_name = "spa_optics_file_"
+      optics_lookup_dir = "/p/lustre2/beydoun1/aerosol_optics"
```

Comment:
This is ok for right now, but is hardcoded for LC machines. We will need to move this file to inputdata and add it to the namelist specification (perhaps in another PR)

Reply:
It might be good to change this in the namelist now.  No one but Hassan will have access to this folder unless he changes the permissions.  @hassanbeydoun , you can follow the example of any of the input file variables in the namelist_definition and namelist_defaults files.  For example, the most recent would be the 'p3_lookup' tables.  You will also need to copy the aerosol optics files to the usr/climdat/... folder so that it is accessible by all.

Reply:
Agreed. It is easy to do and won't be useable by anyone other than Hassan if he doesn't do this.